src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Runtime.InteropServices;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class ModuleBuilderImpl : ModuleBuilder
+    {
+        private readonly Assembly _coreAssembly;
+        private readonly string _name;
+        private readonly Dictionary<string, Type> _coreTypes = new();
+
+        #region Internal Data Members
+
+        internal readonly Dictionary<Assembly, AssemblyReferenceHandle> _assemblyRefStore = new();
+        internal readonly Dictionary<Type, TypeReferenceHandle> _typeRefStore = new();
+        internal readonly List<TypeBuilderImpl> _typeDefStore = new();
+        internal int _nextMethodDefRowId = 1;
+        internal int _nextFieldDefRowId = 1;
+
+        #endregion
```

Comment:
```suggestion
        #region Internal Data Members

        internal readonly Dictionary<Assembly, AssemblyReferenceHandle> _assemblyRefStore = new();
        internal readonly Dictionary<Type, TypeReferenceHandle> _typeRefStore = new();
        internal readonly List<TypeBuilderImpl> _typeDefStore = new();
        internal int _nextMethodDefRowId = 1;
        internal int _nextFieldDefRowId = 1;

        #endregion
```
```suggestion
        private readonly Dictionary<Assembly, AssemblyReferenceHandle> _assemblyRefStore = new();
        private readonly Dictionary<Type, TypeReferenceHandle> _typeRefStore = new();
        private readonly List<TypeBuilderImpl> _typeDefStore = new();
        private int _nextMethodDefRowId = 1;
        private int _nextFieldDefRowId = 1;
```
Can these be private?